kolibri/plugins/epub_viewer/assets/src/views/EpubRendererIndex.vue

```diff
@@ -463,9 +475,22 @@
     methods: {
       updateProgress() {
         if (this.locations.length > 0) {
-          this.$emit('updateProgress', this.summaryTimeSpent / this.expectedTimeToRead);
+          if (this.forceTimeBasedProgress) {
+            // update progress using total time user has spent on the epub
+            this.$emit('updateProgress', this.durationBasedProgress);
+          } else {
+            // update progress using number of pages seen out of available pages
+            this.$emit('updateProgress', this.visitedPages.length / this.locations.length);
```

Comment:
If you did use my suggestion for visitedPages to be an object, you'd change this to `Object.keys(this.visitedPages).length`